### ISSUE TITLE
Broken Permalinks

### ISSUE THREAD

**Issue by pjcforpres2020** [swissspidy/disable-embeds#6]
Hey -

I wanted to give a heads up about an issue I ran into with fancy permalinks breaking due to a PHP error from the following filter:

```php
add_filter( 'rewrite_rules_array', 'disable_embeds_rewrites' );
```

It appears it doesn't break existing fancy permalinks - it just prevents the writing of new ones / changing of the permalinks setting does apply correctly causing 301 redirects to the home page.

Once I removed that line from my theme's functions file fancy permalinks started working.

Server Error Log:

```
Got error 'PHP message: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, function 'disable_embeds_rewrites' not found or invalid function name in /SERVER SPECIFIC DIRECTORIES/public_html/wp-includes/class-wp-hook.php on line 298\n', referer: http://MYURL
```

**Comment by pjcforpres2020** [comment]
Forgot to say - WordPress 4.7.2 on an LEMP PHP7 server and a LAMP PHP 5.4 server.

**Comment by swissspidy** [comment]
Hey there,

Thanks for your report!

I'm a bit confused about that sentence:

> Once I removed that line from my theme's functions file fancy permalinks started working.

Did you just copy the line `add_filter( 'rewrite_rules_array', 'disable_embeds_rewrites' );` from the plugin there? If so, you also need to copy the `disable_embeds_rewrites()` function itself.

However, I do not suggest putting these kinds of things in your theme. Just use the plugin instead.

**Comment by pjcforpres2020** [comment]
Hey -

Thanks for the quick reply.

I have my own "WordPress does stuff I hate" cleanup function I was developing for my new personal starter theme - and in cleaning up my wp_head found this thread discussing oEmbed:

http://wordpress.stackexchange.com/questions/211467/remove-json-api-links-in-header-html/212472

And that would explain while the function was failing - I was thinking it was a WP Core function - but looking at your PHP here it is clearly a function you have defined yourself - and that is not clearly documented in the above link.

When I discovered the problem and solution - I then did a quick Google search to see if anyone else was having issues so I could give a heads up - and that's when I found you had made this plugin / it was what was referenced and linked in the above thread.

Using the plugin you have made does work without issue - since that function is now defined, as it should have been, by your well written and functioning code.

Sorry for any confusion - in fixing this late last night it appeared I had stumbled upon a deprecated function.
